include/triton/Analysis/Allocation.h

```diff
@@ -18,16 +18,39 @@ namespace mlir {
 namespace triton {
 class AllocationAnalysis;
 
-SmallVector<unsigned>
-getScratchConfigForCvtLayout(triton::gpu::ConvertLayoutOp op, unsigned &inVec,
-                             unsigned &outVec);
-SmallVector<unsigned> getScratchConfigForCvtLayout(RankedTensorType srcType,
-                                                   RankedTensorType dstType,
-                                                   unsigned &inVec,
-                                                   unsigned &outVec);
-SmallVector<unsigned> getRepShapeForCvtLayout(triton::gpu::ConvertLayoutOp op);
-SmallVector<unsigned> getRepShapeForCvtLayout(RankedTensorType srcTy,
-                                              RankedTensorType dstTy);
+struct ScratchConfig {
+  SmallVector<unsigned> repShape;
+  SmallVector<unsigned> paddedRepShape;
+  unsigned inVec{1};
+  unsigned outVec{1};
+  unsigned paddedDim;
+  unsigned paddedSize{0};
```

Comment:
Is this bytes or elements?  Is this the size of the one padded dimension, or is it the total size of the padded tensor?  Suggest adding a comment.